editor/src/components/canvas/canvas-strategies/strategies/grid-rearrange-keyboard-strategy.ts

```diff
@@ -0,0 +1,179 @@
+  if (cell == null) {
+    return null
+  }
+
+  const parentGridPath = EP.parentPath(target)
+
+  const grid = MetadataUtils.findElementByElementPath(canvasState.startingMetadata, parentGridPath)
+  if (grid == null) {
+    return null
+  }
+  const gridTemplate = grid.specialSizeMeasurements.containerGridProperties
+
+  const cellBounds = getGridCellBoundsFromCanvas(cell, canvasState.scale, canvasState.canvasOffset)
+  if (cellBounds == null) {
+    return null
+  }
+
+  return {
+    id: 'GRID_KEYBOARD_REARRANGE',
+    name: 'Grid rearrange',
+    descriptiveLabel: 'Grid rearrange',
+    icon: {
+      category: 'modalities',
+      type: 'reorder-large',
+    },
+    controlsToRender: [
+      {
+        control: GridControls,
+        props: { targets: [parentGridPath] },
+        key: GridControlsKey(parentGridPath),
+        show: 'always-visible',
+        priority: 'bottom',
+      },
+    ],
+    fitness: fitness(interactionSession),
+    apply: () => {
+      if (interactionSession == null || interactionSession.interactionData.type !== 'KEYBOARD') {
+        return emptyStrategyApplicationResult
+      }
+      if (
+        gridTemplate.gridTemplateColumns?.type !== 'DIMENSIONS' ||
+        gridTemplate.gridTemplateRows?.type !== 'DIMENSIONS'
+      ) {
+        return emptyStrategyApplicationResult
+      }
+
+      const interactionData = interactionSession.interactionData
+
+      const horizontalDelta = getKeysDelta(interactionData.keyStates, 'horizontal')
+      const verticalDelta = getKeysDelta(interactionData.keyStates, 'vertical')
+
+      let gridProps: Partial<GridElementProperties> = {
+        ...cell.specialSizeMeasurements.elementGridProperties,
+      }
+
+      if (horizontalDelta !== 0) {
+        const { from, to } = getNewBounds(
+          cellBounds.originCell.column + horizontalDelta,
+          gridTemplate.gridTemplateColumns.dimensions.length,
+          cellBounds.width,
+        )
+        gridProps.gridColumnStart = gridPositionValue(from)
+        gridProps.gridColumnEnd = gridPositionValue(to)
+      }
+
+      if (verticalDelta !== 0) {
+        const { from, to } = getNewBounds(
+          cellBounds.originCell.row + verticalDelta,
+          gridTemplate.gridTemplateRows.dimensions.length,
+          cellBounds.height,
+        )
+        gridProps.gridRowStart = gridPositionValue(from)
+        gridProps.gridRowEnd = gridPositionValue(to)
+      }
+
+      return strategyApplicationResult(setGridPropsCommands(target, gridTemplate, gridProps))
+    },
+  }
+}
+
+function fitness(interactionSession: InteractionSession | null): number {
+  if (interactionSession == null || interactionSession.interactionData.type !== 'KEYBOARD') {
+    return 0
+  }
+
+  const accumulatedPresses = accumulatePresses(interactionSession.interactionData.keyStates)
+  const matches = accumulatedPresses.some(
+    (accumulatedPress) =>
+      Array.from(accumulatedPress.keysPressed).some(
```

Comment:
btw a nice trick with the new `Set.prototype.intersection` can be:
```
accumulatedPress.keysPressed.intersection(new Set(['left', 'right', 'up', 'down']).size > 0
```
:)

Reply:
ah nice, I only copy-pasted this from the existing reorder strat :)